### PR TITLE
CBL-5950: Port VS defaults from master and fix isLazy

### PR DESCRIPTION
**Pull request by velicuvlad**
- port a3b7a758b165d96bc3985cee9081737019740f9e
- port 2dad3b185083fdda5cea1c5649130d7953257144
- fix isLazy. should be `false` not `"false"`
- change `DistanceMetric` value from `Euclidean` to `EuclideanSquare`

### CODE REVIEW

File: gen_defaults/cbl-defaults.json [hunk 1]
```diff
@@ -82,6 +92,81 @@
...
+                "type_c": {
+                    "id": "unsigned",
+                    "subset": "system"
+                },
+                "type_objc": {
+                    "id": "unsigned int",
+                    "subset": "system"
+                },
+                "type_csharp": {
+                    "id": "int",
```
Comment (borrrden) [inline gen_defaults/cbl-defaults.json:116]: This is incorrect.  C# should be uint, so this (and C) should be removed as the default generated type is fine.

File: gen_defaults/cbl-defaults.json [hunk 2]
```diff
@@ -82,6 +92,81 @@
...
```
Comment (borrrden) [inline gen_defaults/cbl-defaults.json:141]: Same issue

File: gen_defaults/cbl-defaults.json [hunk 3]
```diff
@@ -82,6 +92,81 @@
...
```
Comment (borrrden) [inline gen_defaults/cbl-defaults.json:166]: Same issue

**Review by borrrden — CHANGES_REQUESTED** [review]

**Review by bmeike — COMMENTED** [review]
I defer to Jim, here

**Comment by velicuvlad** [comment]
I've now also changed `euclidean` to `euclideanSquared` for the upcoming changes

**Review by bmeike — APPROVED** [review]

**Pull request merged**